control/state.py

```diff
@@ -222,6 +227,53 @@ def get_omap_version(self) -> int:
                 f" invalid number of values ({value_list}).")
             raise
 
+    def lock_omap(self, timeout = None):
+        got_lock = False
+
+        if timeout is None:
+            self.logger.warning(f"Can't lock OMAP file without a lock timeout")
+            return False
```

Comment:
I'm uncertain why errors typically generate exceptions, but in this case, a False value is returned for what appears to be a default value of timeout argument. Is it possible to remove the return value for lock_omap(), raise an exception for errors, and have the caller assume that they "got the lock" otherwise?

Reply:
@baum we shouldn't get to this situation. The value of the timeout should be set to something. I didn't want to crash in case in wasn't but still wanted to let the caller know the lock was not taken.

Reply:
I do not like this inconsistency in the interface of using both return values and exceptions. I think the approach of raise exception on errors, void return value would make the caller life easier and the code cleaner.. No need both handle the exception and check return value.